orchagent/pfcwdorch.cpp

```diff
@@ -446,173 +392,110 @@ bool PfcWdSwOrch<DropHandler, ForwardHandler>::startWdOnQueue(sai_object_id_t qu
-            {
-                stormedQueues.push_back(sai_serialize_object_id(queueId));
-            }
-            // Queue is not stormed
-            else
-            {
-                normalQueues.push_back(sai_serialize_object_id(queueId));
-            }
-        }
-    }
+    string queueIdStr;
+    string event;
+    vector<swss::FieldValueTuple> values;
 
-    // Run scripts for selected queues to see if their state changed
-    // from normal to stormed and vice versa
-    set<string> stormCheckReply;
-    set<string> restoreCheckReply;
-    vector<string> argv =
-    {
-        to_string(COUNTERS_DB),
-        COUNTERS_TABLE,
-        to_string(nearestTime * 1000)
-    };
+    wdNotification.pop(queueIdStr, event, values);
 
-    if (!normalQueues.empty())
-    {
-        stormCheckReply = runRedisScript(db, detectSha, normalQueues, argv);
-    }
+    sai_object_id_t queueId = SAI_NULL_OBJECT_ID;
+    sai_deserialize_object_id(queueIdStr, queueId);
 
-    if (!stormedQueues.empty())
+    auto entry = m_entryMap.find(queueId);
+    if (entry == m_entryMap.end())
     {
-        restoreCheckReply = runRedisScript(db, restoreSha, stormedQueues, argv);
+        SWSS_LOG_ERROR("Queue %s is not registered", queueIdStr.c_str());
+        return;
     }
 
-    // Update internal state of queues and their time
-    for (auto& queueKv : m_entryMap)
+    if (event == "storm")
     {
-        sai_object_id_t queueId = queueKv.first;
-        PfcWdQueueEntry& queueEntry = queueKv.second;
-
-        string queueIdStr = sai_serialize_object_id(queueId);
-
-        // Queue became stormed
-        if (stormCheckReply.find(queueIdStr) != stormCheckReply.end())
+        if (entry->second.action == PfcWdAction::PFC_WD_ACTION_DROP)
         {
-            if (queueEntry.c_action == PfcWdAction::PFC_WD_ACTION_DROP)
-            {
-                queueEntry.handler = make_shared<DropHandler>(
-                        queueEntry.portId,
-                        queueId,
-                        queueEntry.index,
-                        PfcWdOrch<DropHandler, ForwardHandler>::getCountersTable());
-            }
-            else if (queueEntry.c_action == PfcWdAction::PFC_WD_ACTION_FORWARD)
-            {
-                queueEntry.handler = make_shared<ForwardHandler>(
-                        queueEntry.portId,
-                        queueId,
-                        queueEntry.index,
-                        PfcWdOrch<DropHandler, ForwardHandler>::getCountersTable());
-            }
-            else
-            {
-                throw runtime_error("Invalid PFC WD Action");
-            }
-
-            queueEntry.pollTimeLeft = queueEntry.c_restorationTime;
+            entry->second.handler = make_shared<DropHandler>(
+                    entry->second.portId,
+                    entry->first,
+                    entry->second.index,
+                    PfcWdOrch<DropHandler, ForwardHandler>::getCountersTable());
```

Comment:
Can we write all PFCWD state back to PFC_WD_STATE_TABLE (including queue status, drop_pkts etc) in phase 3?